shared/common-adapters/toast.native.js

```diff
@@ -8,31 +8,43 @@ import type {Props} from './toast'
 
 type State = {
   opacity: NativeAnimated.Value,
+  visible: boolean,
 }
 class Toast extends React.Component<Props, State> {
-  state = {opacity: new NativeAnimated.Value(0)}
+  state = {opacity: new NativeAnimated.Value(0), visible: false}
+  timeoutID: TimeoutID
 
   componentDidUpdate(prevProps: Props) {
     if (this.props.visible && !prevProps.visible) {
-      NativeAnimated.timing(this.state.opacity, {
-        duration: 50,
-        easing: NativeEasing.linear,
-        toValue: 1,
-      }).start()
+      this.setState({visible: true}, () =>
+        NativeAnimated.timing(this.state.opacity, {
+          duration: 100,
+          easing: NativeEasing.linear,
+          toValue: 1,
+        }).start()
+      )
     }
     if (!this.props.visible && prevProps.visible) {
       NativeAnimated.timing(this.state.opacity, {
-        duration: 50,
+        duration: 100,
         easing: NativeEasing.linear,
         toValue: 0,
       }).start()
+      this.timeoutID = setTimeout(() => this.setState({visible: false}), 100)
     }
   }
 
+  componentWillUnmount() {
```

Comment:
we should use timerhoc and not do this outselves

Reply:
done 👍